iceoryx2-cal/Cargo.toml

```diff
@@ -26,6 +26,7 @@ iceoryx2-bb-memory = { workspace = true }
 iceoryx2-bb-lock-free = { workspace = true }
 iceoryx2-pal-concurrency-sync = { workspace = true }
 
+dirs = { workspace = true }
```

Comment:
Please remove this and the `config_path` component from `iceoryx2_cal`. This is not a iceoryx2 concept. The mistake is on our side since we did not yet had the time to document the iceoryx2 concept pattern in detail, what it is, what problem it solves and how it shall be implemented.

A concept is an abstract mechanism:
1. That has multiple implementations.
2. Can be constructed and used via the traits of the concepts
3. Comes with a generic test suite to verify the same behavior of all concept implementations.

One example is the communication channel.
 * it has a receiver and a sender
 * it comes with a builder trait
 * the sender sends data, the receiver receives data
 * it can be implemented by a UNIX domain socket, message queue, named pipe or a shared memory queue
 * it comes with a test suite that verifies that the sent data can be retrieved with the corresponding receiver.

I agree that you require a way of acquiring the config directory and your approach seems valid but this should be part of the `iceoryx2/src/config.rs`